lib/passageidentity/user_api.rb

```diff
@@ -108,105 +86,71 @@ def deactivate(user_id:)
       updates['phone'] = phone unless phone.empty?
       updates['user_metadata'] = user_metadata unless user_metadata.empty?
-      begin
-        response = @user_client.update_user(@app_id, user_id, updates, @req_opts)
-        response.user
-      rescue Faraday::Error => e
-        if e.is_a? Faraday::ResourceNotFound
-          raise PassageError.new(
-            message: "Passage User with ID \"#{user_id}\" does not exist",
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        else
-          raise PassageError.new(
-            'failed to update Passage User',
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        end
-      end
+
+      update_v2(user_id: user_id, options: updates)
     end
 
     def create(email: '', phone: '', user_metadata: {})
+      warn '[DEPRECATED] the `create` method parameters will change to `args: CreateUserArgs`.' \
+      'Parameters will change on or after 2025-1.'
+
       create = {}
       create['email'] = email unless email.empty?
       create['phone'] = phone unless phone.empty?
       create['user_metadata'] = user_metadata unless user_metadata.empty?
-      begin
-        response = @user_client.create_user(@app_id, create, @req_opts)
-        response.user
-      rescue Faraday::Error => e
-        raise PassageError.new(
-          'failed to create Passage User',
-          status_code: e.response[:status],
-          body: e.response[:body]
-        )
-      end
+
+      create_v2(args: create)
     end
 
     def delete(user_id:)
       user_exists?(user_id)
 
       begin
         @user_client.delete_user(@app_id, user_id, @req_opts)
-        true
       rescue Faraday::Error => e
-        if e.is_a? Faraday::ResourceNotFound
-          raise PassageError.new(
-            "passage User with ID \"#{user_id}\" does not exist",
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        else
-          raise PassageError.new(
-            'failed to delete Passage User',
-            status_code: e.response[:status],
-            body: e.response[:body]
-          )
-        end
+        raise PassageError.new(
+          'failed to delete Passage User',
+          status_code: e.response[:status],
+          body: e.response[:body]
+        )
       end
     end
 
-    def delete_device(user_id:, device_id:)
+    def revoke_device(user_id:, device_id:)
       user_exists?(user_id)
       device_exists?(device_id)
 
       begin
         @user_device_client.delete_user_devices(@app_id, user_id, device_id, @req_opts)
-        true
       rescue Faraday::Error => e
         raise PassageError.new(
-          'failed to delete Passage User Device',
           status_code: e.response[:status],
           body: e.response[:body]
         )
       end
     end
 
+    def delete_device(user_id:, device_id:)
```

Comment:
this still needs to return true since this method already existed and returned true before, while the new `revoke_device` can stay as it is without returning anything

Reply:
Same here as above. Does the successful `@user_device_client.delete_user_devices` return true?